### ISSUE TITLE
Specifying huge "max nodes" value, consumes a lot of memory

### ISSUE THREAD

**Issue by korniltsev** [grafana/pyroscope#2280]
![Screenshot_20230819_093552](https://github.com/grafana/pyroscope/assets/331773/085cc685-cd68-4f3f-9f2a-0496e2b66b64)

maybe we should cap it ?

**Comment by kolesnikovae** [comment]
Yeah, I think the natural limit is the number of nodes in the tree. Fix: #2281
